util/build.sh

```diff
@@ -83,7 +83,19 @@ get_options() {
 main() {
     get_options "$@"
 
-    sudo docker run --rm -w /curve --user $(id -u ${USER}):$(id -g ${USER}) -v $(pwd):/curve -v ${HOME}:${HOME} -v /etc/passwd:/etc/passwd:ro -v /etc/group:/etc/group:ro -v /etc/shadow:/etc/shadow:ro --privileged opencurvedocker/curve-base:build-$g_os bash util/build_in_image.sh "$@"
+    sudo docker run \
+        --rm \
+        -w /curve \
+        --user $(id -u ${USER}):$(id -g ${USER}) \
+        -v $(pwd):/curve \
+        -v ${HOME}:${HOME} \
+        -v /etc/passwd:/etc/passwd:ro \
+        -v /etc/group:/etc/group:ro \
+        -v /etc/shadow:/etc/shadow:ro \
+        -e BUILD_OPTS="${BUILD_OPTS}" \
+        --privileged \
+        opencurvedocker/curve-base:build-$g_os \
+        bash util/build_in_image.sh "$@"
```

Comment:
Can the docker related stuff be organized and put in util/docker.sh? for reference only

Reply:
Of course. I will add a `docker.sh`

Reply:
Do you mean we just add the docker-related stuff to `util/docker.sh` without calling them? Or we add and call it from other scripts?

I have assumed the latter one. After some investigation, I think I will need to edit `docker.sh` to an extent that I think is too much to make it flexible enough to call in other scripts. Provided it already have some useful contents, maybe I will leave `docker.sh` as is.

Reply:
Hi I organized common docker-related arguments into a separate file called `docker_opts.sh`. It should be easier to manage all docker things from one place now. 

I only extracted common docker-related arguments, so other scripts can source them and make modifications freely. The reason why I did not use `util/docker.sh` is that if I edit it to be flexible engough to be used in other scripts, I am afraid to break other things that are already using this script.